x-pack/plugins/data_enhanced/server/search/session/get_search_status.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ApiResponse } from '@elastic/elasticsearch';
+import { ElasticsearchClient } from 'src/core/server';
+import { isAsyncSearchStatusResponse, SearchStatus } from './types';
+import { AsyncSearchResponse } from '../types';
+import { BackgroundSessionSearchInfo } from '../../../common';
+
+export async function getSearchStatus(
+  client: ElasticsearchClient,
+  asyncId: string
+): Promise<Pick<BackgroundSessionSearchInfo, 'status' | 'error'>> {
+  try {
+    // TODO: Handle strategies other than KQL
+    const response: ApiResponse<AsyncSearchResponse> = await client.asyncSearch.status({
+      id: asyncId,
+    });
+    if (isAsyncSearchStatusResponse(response)) {
```

Comment:
When would we get something other than an error or an `AsyncSearchStatusResponse` here? Maybe instead of this check we can wrap this in a try/catch and then replace above with this:

```ts
    const response: ApiResponse<AsyncSearchStatusResponse> = await client.asyncSearch.status({
      id: asyncId,
    });
```